tests/EnumTest.php

```diff
@@ -405,6 +405,35 @@ function foo(): int|Code|null
                 'ignored_issues' => [],
                 'php_version' => '8.1',
             ],
+            'EnumCaseReconciliation' => [
+                'code' => '<?php
+                    enum Code: int
+                    {
+                        case Ok = 0;
+                        case Fatal = 1;
+                    }
+
+                    function foo(): Code|null
+                    {
+                        return null;
+                    }
+
+                    $code = foo();
+                    $code1 = null;
+                    $code2 = null;
+                    if($code instanceof Code){
+                        $code1 = $code;
+                    }
+                    if(!$code instanceof Code){
+                        $code2 = $code;
+                    }',
```

Comment:
Does it actually test reconciliation with `Enum|Enum::Case`? It doesn't seem like an individual enum case would ever appear in this test case.

Reply:
what I wanted to test was the enum case not leaking to the negated if like in here: https://github.com/vimeo/psalm/issues/7654#issuecomment-1037378155

The test would be cleaner without null values, (it showed Code|Code::Ok and never on the other var before) but assertions doesn't work if the var appears only in an if

Reply:
I mean, this test passes on 5.0a1 (I just tested that). So it doesn't cover changes introduced in this PR.

Reply:
Ugh :( I'll try to design a better test then!

Reply:
What you could do instead is to abort (exit or throw) in the `if` body, and then assert the type as it was changed below the `if`. E.g.
```php
[
            'EnumCaseReconciliation' => [
                'code' => '<?php
                    enum Code: int
                    {
                        case Ok = 0;
                        case Fatal = 1;
                    }

                    function foo(): Code|null|int
                    {
                        return null;
                    }

                    $code = foo() ?? Code::Ok;
                    if (!$code instanceof Code) throw new RuntimeException;
                    ',
                'assertions' => [
                    '$code===' => 'Code',
                ],
                'ignored_issues' => [],
                'php_version' => '8.1',
            ],
]
```